test/scheduler/usecases/customer/update_turn_test.js

```diff
@@ -1,15 +1,320 @@
+const sinon = require('sinon');
 const { expect, assert } = require('chai');
+const tk = require('timekeeper');
 
-// const { BranchStoreMock } = require('./mocks');
-const customerErrors = require('../../../../scheduler/usecases/customer/errors');
-const UpdateTurn = require('../../../../scheduler/usecases/customer/update-turn');
+const Turn = require('../../../../scheduler/turn');
+const Branch = require('../../../../scheduler/branch');
+const Schedule = require('../../../../scheduler/schedule');
+const Customer = require('../../../../scheduler/customer');
+const BranchStore = require('../../../../scheduler/stores/branch');
+const TurnStore = require('../../../../scheduler/stores/turn');
+const CustomerStore = require('../../../../scheduler/stores/customer');
+const storeErrors = require('../../../../scheduler/stores/errors');
+const useCaseErrors = require('../../../../scheduler/usecases/customer/errors');
+const CustomerUpdateTurn = require('../../../../scheduler/usecases/customer/update-turn');
 
 
 suite('Use Case: Customer updates turn', () => {
 
-  setup(() => {});
+  setup(() => {
+    sandbox = sinon.createSandbox();
+    schedule = new Schedule();
+    branch = new Branch({
+      id: 'restaurant-branch-id',
+      schedule: schedule,
+    });
+    customer = new Customer({
+      id: 'customer-id',
+    });
+    turn = new Turn({
+      id: 'turn-id',
+      name: 'Test',
+      branch: branch,
+      customer: customer,
+    });
+    newTurn = new Turn({
+      guests: 4,
+      name: 'Update Test',
+    });
 
-  teardown(() => {});
+    customerStore = new CustomerStore();
+    branchStore = new BranchStore();
+    turnStore = new TurnStore();
+  });
 
-  test('', () => {});
+  teardown(() => {
+    sandbox.restore();
+    tk.reset();
+  });
+
+  test('customer updates a turn', async () => {
+    const requestedTime = new Date(Date.UTC(2018, 12, 9));
+    const updatedTurn = new Turn({
+      name: newTurn.name,
+      guests: newTurn.guests,
+      requestedTime: requestedTime,
+      customer: customer,
+      branch: branch,
+    });
+
+    sandbox.stub(branchStore, 'find')
+      .returns(Promise.resolve(branch));
+    sandbox.stub(customerStore, 'find')
+      .returns(Promise.resolve(customer));
+    sandbox.stub(turnStore, 'find')
+      .returns(Promise.resolve(turn));
+    sandbox.stub(turnStore, 'update')
+      .returns(Promise.resolve(updatedTurn));
+    sandbox.stub(Branch.prototype, 'isOpen')
```

Comment:
Since `branch` object is created during the execution, we do not have a reference to it. This is equivalent to create a stub for `any` branch instance.